planner/core/common_plans.go

```diff
@@ -235,6 +240,18 @@ func (e *Execute) OptimizePreparedPlan(ctx context.Context, sctx sessionctx.Cont
 	}
 	prepared := preparedObj.PreparedAst
 	vars.StmtCtx.StmtType = prepared.StmtType
+	// RC isolation level or statement with for-update read need to check if the schema is changed
+	// before use the cached plan.
+	checkLatestSchema := preparedObj.ForUpdateRead || vars.IsIsolation(ast.ReadCommitted)
+	checkLatestSchema = checkLatestSchema && vars.ConnectionID > 0
+	if checkLatestSchema {
+		is = domain.GetDomain(sctx).InfoSchema()
+		is = temptable.AttachLocalTemporaryTableInfoSchema(sctx, is)
+		txnManager := sessiontxn.GetTxnManager(sctx)
+		if p, isOK := txnManager.GetContextProvider().(*legacy.SimpleTxnContextProvider); isOK {
+			p.InfoSchema = is
+		}
+	}
```

Comment:
Would this check affect the performance? I'm not sure whether the operation of getting the current `information-schema` is heavy or not.

Reply:
Getting the current schema will not affect performance because it actually loads from the cache.